locust/runners.py

```diff
@@ -180,11 +184,23 @@ def kill_locust_greenlets(self, greenlets):
         else:
             for g in greenlets:
                 self.locusts.killone(g)
-    
+        
+    def cpucheck(self):
+        process = psutil.Process()
+        while True:
+            if not self.cpu_threshold_exceeded: 
+                current_cpu = process.cpu_percent()
+                if current_cpu > 90:
+                    self.cpu_threshold_exceeded = True
+                    logging.warning("Loadgen CPU usage above 90%! This may constrain your throughput and even give inconsistent response time measurements! See https://docs.locust.io/en/stable/running-locust-distributed.html for how to distribute the load over multiple CPU cores or machines")
+            gevent.sleep(2.0)
```

Comment:
Maybe we should increase the interval to something like 10 seconds? That would prevent a warning if you have a single task that does something really CPU heavy for a few seconds, and it shouldn't really matter if you get a warning after 2 or 10 seconds. 

Also, do we know how much overhead (if any) `Process.cpu_percent()` introduces?

Reply:
I was thinking about the interval, but decided against increasing it, because even a short spike can cause incorrect measurements. I was thinking it is better to warn a few times too many than a few times too few...

I dont know exactly how much overhead it introduces, but I did a few tests and the call itself never took more than 0.2ms to run on my 2018 MacBook pro. I think it just reading a counter somewhere...

If you want, I can bump the interval to 5 seconds.

Reply:
Ok, five seconds sounds good!